ext/mysqlnd/mysqlnd_ps_codec.c

```diff
@@ -241,7 +242,21 @@ ps_fetch_time(zval * zv, const MYSQLND_FIELD * const field, const unsigned int p
 		t.time_type = MYSQLND_TIMESTAMP_TIME;
 	}
 
-	length = mnd_sprintf(&value, 0, "%s%02u:%02u:%02u", (t.neg ? "-" : ""), t.hour, t.minute, t.second);
+    if (field->decimals > 0 && field->decimals < 7) {
+        length = mnd_sprintf(
+            &value,
+            0,
+            "%s%02u:%02u:%02u.%0*" ZEND_ULONG_FMT_SPEC,
```

Comment:
The field width is not needed anymore, as the argument was truncated.

Reply:
Am I missing something? I thought it still needed to be zero-padded when smaller than 10^(decimals-1).
The leading zero though seems superfluous indeed.